app/src/main/java/exh/recs/BrowseRecommendsScreen.kt

```diff
@@ -29,20 +36,37 @@ class MangaDexSimilarScreen(val mangaId: Long, val sourceId: Long) : Screen() {
             return
         }
 
-        val screenModel = rememberScreenModel { MangaDexSimilarScreenModel(mangaId, sourceId) }
+        val context = LocalContext.current
         val navigator = LocalNavigator.currentOrThrow
 
-        val onMangaClick: (Manga) -> Unit = {
-            navigator.push(MangaScreen(it.id, true))
+        val screenModel = rememberScreenModel {
+            BrowseRecommendsScreenModel(mangaId, sourceId, recommendationSourceName)
         }
-
         val snackbarHostState = remember { SnackbarHostState() }
 
+        val onClickItem = { manga: Manga ->
+            navigator.push(
+                when (isExternalSource) {
+                    true -> SourcesScreen(SourcesScreen.SmartSearchConfig(manga.ogTitle))
+                    false -> MangaScreen(manga.id, true)
+                },
+            )
+        }
+
+        val onLongClickItem = { manga: Manga ->
+            when (isExternalSource) {
+                true -> WebViewActivity.newIntent(context, manga.url, title = manga.title).let(context::startActivity)
+                false -> onClickItem(manga)
+            }
+        }
+
         Scaffold(
             topBar = { scrollBehavior ->
+                val recSource = screenModel.recommendationSource
```

Comment:
```suggestion
                val recSource = remember { screenModel.recommendationSource }
```
Since this uses a get() this needs to be remembered. I will merge it though since this is my only nitpick